Grass Gro tile orientation fix

nova2053 opened the pull request:
<!-- The text between the arrows are comments - they won't be visible on your PR. -->
<!-- To label this PR, add the label(s) without the prefixes surrounded by brackets anywhere, for example: [LABEL] -->
<!-- PRs should at least have one area (A-) label and at least one category (C-) label -->
[bug][game objects]
## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Reverts grass-gro-afflicted tiles back to their roundstart dirs and materials when they're replaced


## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Currently, applying Grass Gro to tiles and then removing it via crowbar results in the tile in question being rotated and often the wrong material. This should fix that.

nova2053 commented:
Demonstration https://www.youtube.com/watch?v=D_BQofifNqI

ZeWaka reviewed (commented):
The problem with this fix is that then all of the grass uses the exact same sprite. The reason why the direction is being set is so it picks between one of the 4 variant icons

nova2053 commented:
> The problem with this fix is that then all of the grass uses the exact same sprite. The reason why the direction is being set is so it picks between one of the 4 variant icons

The sprites for the grass in this PR are still random, just rather hard to see. All three grass tiles with tiny black arrows pointing to them are different from each other, for example
![Screenshot (5784)](https://github.com/user-attachments/assets/ad6a5924-2a9e-44b9-93b2-0dac40ba9fd2)

ZeWaka commented:
> > The problem with this fix is that then all of the grass uses the exact same sprite. The reason why the direction is being set is so it picks between one of the 4 variant icons
> 
> The sprites for the grass in this PR are still random, just rather hard to see. All three grass tiles with tiny black arrows pointing to them are different from each other, for example

If this worked, what are the additional changes for?

Tarmunora commented:
> The sprites for the grass in this PR are still random, just rather hard to see. 

Are those actually random, or just inheriting the dir of the carpet underneath?

nova2053 commented:
> > > The problem with this fix is that then all of the grass uses the exact same sprite. The reason why the direction is being set is so it picks between one of the 4 variant icons
> > 
> > 
> > The sprites for the grass in this PR are still random, just rather hard to see. All three grass tiles with tiny black arrows pointing to them are different from each other, for example
> 
> If this worked, what are the additional changes for?

As it turns out it was just inheriting the previous tile dir, meaning in rooms with largely the same tile directions (such as the tiles in cogmap2 bar), all grass tiles would look the same. Only reason i didn't realize this sooner was because I was testing this in rooms with largely varied tile dirs (cogmap2 chapel and bridge)

nova2053 commented:
> > The sprites for the grass in this PR are still random, just rather hard to see.
> 
> Are those actually random, or just inheriting the dir of the carpet underneath?

They were in fact just inheriting the dir of the carpet underneath, fixed in last commit

nova2053 commented:
Changed the grass to use different icon states rather than dirs, eliminating need for tile rotation. Also removed the bespoke roundstart_step_material and replaced it with initial(step_material)

Tarmunora commented on code/turf/floors.dm:
```diff
@@ -1582,22 +1582,21 @@ TYPEINFO(/turf/simulated/floor/grass)
+	src.icon_state = pick("grass_0", "grass_1", "grass_2", "grass_3")
 	src.UpdateIcon()
 	if(prob(30))
-		src.icon_state += pick("_p", "_w", "_b", "_y", "_r", "_a")
+		src.icon_state = pick("grass_p", "grass_w", "grass_b", "grass_y", "grass_r", "grass_a")
```
flower sprites will be unidirectional

nova2053 replied:
Flower sprites will probably still inherit direction from the tiles beneath them, and there's a bunch of different flower colours too, so it probably won't be too noticeable even in the case they do happen to be the same orientation

nova2053 replied:
Alternative solution: Rotate tiles only when flower sprite is selected, though this seems like it would render the separation of grass icons pointless as it would require the tile to be rotated back anyways

Tarmunora commented:
eh, this is... probably good enough
